### PR TITLE
Split Copilot docs and add top-level node in ToC

### LINKED ISSUE #6916: Copilot top level section in docs
* Overview (talk about what it is about, how can people use it, drive them to acquisition)
* Best Practices (when to use regular Copilot, some minor customizations, how to use side bar, when to use inline, generate commit message, generate PR titles, )
* Advanced (/docs, /fix, references, etc)
* Extensibility docs ( @isidorn can drive this)

image

### PR DESCRIPTION
**Pull request by ntrogh**
Fixes #6916

### CODE REVIEW

**Review by sandy081 — DISMISSED** [review]

**Comment by isidorn** [comment]
@ntrogh looks good -> approved. However I would not put the Extensibility section under the Copilot part. I think it belongs more under the API part of the docs. Maybe under Extension Guides there is an Copilot Agent section or something like that. What do you think?

And I noticed you have conflicts in this PR.

**Review by isidorn — DISMISSED** [review]

**Comment by ntrogh** [comment]
> @ntrogh looks good -> approved. However I would not put the Extensibility section under the Copilot part. I think it belongs more under the API part of the docs. Maybe under Extension Guides there is an Copilot Agent section or something like that. What do you think?
> 
> And I noticed you have conflicts in this PR.

Thanks for reviewing and agree that extensibility would fit better in API section. The extensibility doc is not visible in the ToC yet, but I'll move it under API. 

Will look into the merge conflict.

File: docs/copilot/advanced.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+![Inline chat /doc example to generate documentation code comments for a calculator class](images/advanced/inline-chat-doc-example.png)
+
+## Generating unit tests
+
+The `@workspace /tests` slash command enables you to generate unit tests for your code. When generating the tests, the agent can detect the testing framework you're using and generates new tests in the same style. When you use inline chat, you can use the `/tests` slash command.
+
+![Chat view showing the /tests slash command to generate unit tests for the Divide method in a C# calculator class](images/advanced/workspace-agent-tests-example.png)
+
+When inline chat proposes creating new files, such as when using the `/tests` slash command, you can choose the file name and location by selecting **Create As** from the **Create** drop down.
```
Comment (ntrogh) [inline docs/copilot/advanced.md:84]: Should we add a section about getting help with failed terminal commands and cross references from here: https://github.com/ntrogh/vscode-docs/blob/c110423efc2b474d97bbc2c242cfc101e27d244a/docs/terminal/shell-integration.md#L115-L118

File: docs/copilot/advanced.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+* @workspace /explain: Explain step-by-step how the selected code works.
+* @workspace /fix: Propose a fix for the bugs in the selected code.
+* @workspace /new: Create a new project based on a natural language description.
+* @workspace /newNotebook: Create a new Jupyter Notebook based on your description.
+* @workspace /tests: Generate unit tests for the selected code.
+* @vscode /api: Questions about VS Code extension development.
+* @vscode /search: Generate query parameters for workspace search.
+* @terminal: Explain how to do something in the integrated terminal.
+
```
Comment (ntrogh) [inline docs/copilot/advanced.md:42]: Should we add a section about suggested fixes for failed terminal commands? We can then refer from the terminal docs: https://github.com/ntrogh/vscode-docs/blob/c110423efc2b474d97bbc2c242cfc101e27d244a/docs/terminal/shell-integration.md#L115-L118

Comment (isidorn) [reply docs/copilot/advanced.md:42]: I think we can have a terminal section, and start small with suggested fixes.
We might add more terminal copilot integration in the future, so it is good to have an open-ended section.

fyi @Tyriar @meganrogge

**Review by isidorn — DISMISSED** [review]
@ntrogh this looks good to me!
I would love to be able to review these changes in a pre-publish website. Would just be easier for me to figure out where each page goes.

Can you please ping me once we have this pre-published, so I give it another review.

Thank you!

**Comment by ntrogh** [comment]
@isidorn requesting a final review before publishing the first iteration of the updated Copilot docs.

File: docs/copilot/ai-powered-suggestions.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+1. [Getting started](/docs/copilot/getting-started.md): set up VS Code for Copilot and get your first AI-powered suggestions
+
+1. [Getting started with Copilot Chat](/docs/copilot/getting-started-chat.md): install the Copilot Chat extension and have your first AI chat conversations
+
+## Using Copilot
+
+Now that you've signed up for Copilot and activated the extension, let's see its assistance in action!
+
+GitHub Copilot provides suggestions for numerous languages and a wide variety of frameworks, and it works especially well for Python, JavaScript, TypeScript, Ruby, Go, C# and C++.
```
Comment (isidorn) [inline docs/copilot/ai-powered-suggestions.md:37]: Nit: Python, JavaScript, TypeScript, Ruby, Go, C# and C++.

This language ordering is weird. Ruby being very high.
I suggest to just use top languages in VS Code.

We probably also have telemetry that shows top copilot languages

fyi @bamurtaugh 

Feel free to publish, and the later we can revisit.

File: docs/copilot/ai-powered-suggestions.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+## Chat features
+
+In addition to inline suggestions, you can also get assistance from Copilot via a chat interface. Copilot Chat is supported in several ways:
+
+* **Chat view:** Ask Copilot for help with any task or question in the GitHub Copilot Chat view.
+* **Inline Chat:** Talk with Copilot while writing code, inline in your files.
+* **Quick Chat:** Bring up a Chat dropdown for quick questions and suggestions.
+
+### Install the GitHub Copilot Chat extension
```
Comment (isidorn) [inline docs/copilot/ai-powered-suggestions.md:79]: The Chat extension comes as part of Copilot.
I am not sure if we need to call this out. This extension is an implementation detail. Maybe we could remove this section. So my point is that all Copilot users will already have it installed.

fyi @bamurtaugh

File: docs/copilot/ai-powered-suggestions.md (hunk 3 above)
Comment (ntrogh) [reply docs/copilot/ai-powered-suggestions.md:37]: I kept this fragment from the original article. Makes sense, especially as this list might evolve over time.

File: docs/copilot/faq.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+DateApproved: 02/1/2024
+MetaDescription: Frequently asked questions for using GitHub Copilot in Visual Studio Code.
+---
+# GitHub Copilot frequently asked questions
+
+This article answers frequently asked questions about using GitHub Copilot in Visual Studio Code.
+
+## General
+
+### I can't find Copilot Chat in the Activity Bar
```
Comment (isidorn) [inline docs/copilot/faq.md:16]: Not buying this question. Are users really asking this?

**Review by isidorn — APPROVED** [review]

File: docs/copilot/ai-powered-suggestions.md (hunk 4 above)
Comment (bamurtaugh) [reply docs/copilot/ai-powered-suggestions.md:79]: I think calling this out as an implementation detail is a good idea - there's been user feedback that it's confusing why there are multiple extensions. But I agree with what @isidorn mentions that maybe this shouldn't be an entire section, just a note in one of the earlier sections that installing the main Copilot extension will give you both Copilot completions and Copilot Chat.

**Pull request merged**